### ISSUE TITLE
"Wrong password" if multiple copies of keyfile in keystore-folder

### ISSUE THREAD

**Issue by yonilevy** [ethereum/mist#780]
Node type: geth
OS: osx
Network type: main
Mist version: 0.7.4

I've upgraded by Ethereum-Wallet OSX app from 0.7.2 to 0.7.4, right after making a successful transaction. I tried making another transaction, copy pasting the exact same password I've used before (both were copied from a clipboard manager, so they're *exactly* the same). In 0.7.4 I get the "Wrong Password" message.

I've checked the `node.log` file, nothing related there.

The console says `Error: Transaction denied`, with a bunch of network errors connecting to `cryptocompare.com` (which I really hope aren't related). Here's a screenshot:

![image](https://cloud.githubusercontent.com/assets/31284/15808761/aa526b7e-2b88-11e6-9c77-16497f75eb51.png)

I went back to 0.7.2 and re-tested--  **it works fine there**. It got broken somewhere between (0.7.2 and 0.7.4].

Might be worth noting my password contains some special characters including: `$%&`.

Would love to further assist in resolving this issue.

**Comment by alexvandesande** [comment]
We are changing a lot of the node connection on 0.7.5 so it might be resolved there. Please check back on the next release

**Comment by yonilevy** [comment]
@alexvandesande thanks, i'll update when 0.7.5 is out

**Comment by alexvandesande** [comment]
0.7.5 is out. Please update and open again if it persists.

**Comment by yonilevy** [comment]
@alexvandesande tested on 0.7.5, still getting Wrong Password :(

**Comment by igetgames** [comment]
I've tried 0.7.5 for the first time, and after several successful transactions and attempting a new contract execution I get "Wrong Password". Instead of my wallet contract being shown as the sender and the contract I'm attempting to execute on being shown as the receiver, the _last_ transaction's sender and receiver fields show up, and the Ether amount is 0.00 (I'm attempting to send 1500 Finney). I don't see any console errors in the developer tools.

It's alarming that I've been using the same password for every transaction (copied automatically out of my password manager), and I can't unlock the wallet contract from geth either. I'll try downgrading to 0.7.2.

**Comment by igetgames** [comment]
That was a bust. After loading 0.7.2, my watched and wallet contracts are gone. The wallet contract, the one holding my either, was the one whose password all of the sudden stopped working. Switched back to 0.7.4 and it's the same. My Ether's now in the ether. Fuck.

Can anyone suggest an alternative Ethereum client? At least this one was pretty.

**Comment by igetgames** [comment]
Is there any way I can attempt to recover the vanished wallet contract?

**Comment by luclu** [comment]
@igetgames you can reimport the wallet contract. Simply chose to create a new one and change its type to import. If you don't remember the address use a blockchain-explorer to look it up following your accounts transactions.

paging @hiddentao (instance of mising db entries)

**Comment by alexvandesande** [comment]
@yonilevy I had a similar issue with you and was able to fix it. Can you please check your keystore folder and see if you have multiple copies of the same key there? Try to move out all keys but the one you want to use

**Comment by luclu** [comment]
@alexvandesande I can confirm this reproduces the behaviour for me.

**Comment by igetgames** [comment]
This might be my issue:
https://www.reddit.com/r/ethereum/comments/4ngvb7/reminder_even_if_using_a_wallet_contract_be_sure/

I'm sending some coins from another wallet to my main account to see if my wallet contract shows up. I don't have any Ether in my main account so I don't have the button to create a new wallet contract. I have a couple of UX suggestions in that regard, I'll report those separately.

**Comment by alexvandesande** [comment]
@igetgames please post this in a separate thread, this is about the wrong password bug

**Comment by yonilevy** [comment]
@alexvandesande i indeed had two duplicated keys/accounts. removing one didn't immediately solve the issue, but after a reboot it now works!

**Comment by igetgames** [comment]
Yeah. If you have a 0 ETH balance in your main account, and you attempt to send funds to an external address from a wallet contract then you will be presented with a screen attempting to send a 0 ETH balance from your main account to your wallet contract, instead of the address you were attempting to send to. If enter your password and attempt to send, you get a "Wrong Password" message. Should I file those steps as a separate wrong password bug?

**Comment by luclu** [comment]
@igetgames your issue is not related to this one. Yours has its own thread here: https://github.com/ethereum/mist/issues/480

**Comment by igetgames** [comment]
Fair enough. I don't get any info telling me that this is the problem, so the only information I had to go off of was the Wrong Password message. I'll file an issue for that flow separately. Thanks!

**Comment by luclu** [comment]
@alexvandesande geth (>1.3.6) doesn't allow for multiple copies of the same keyfile. We might follow this rationale and handle this error.
```
> personal.unlockAccount("6e4cabb35dbcdbe12e9389d0a91b7649da7427ff")
Unlock account 6e4cabb35dbcdbe12e9389d0a91b7649da7427ff
Passphrase:
multiple keys match address (/Users/ich/Library/Ethereum/keystore/UTC--2016-05-13T14-42-45.275876516Z--6e4cabb35dbcdbe12e9389d0a91b7649da7427ff, /Users/ich/Library/Ethereum/keystore/UTC--2016-05-13T14-42-45.275876516Z--6e4cabb35dbcdbe12e9389d0a91b7649da7427ff Kopie)
```

**Comment by molecular** [comment]
I'm running into this in 0.7.6, worked fine in 0.7.5

**Comment by luclu** [comment]
An alert has been added to the wallet: https://github.com/ethereum/mist/issues/664.

**Comment by luclu** [comment]
Fixed in `0.8.1`. Please update.
